samples/openthread/ncp/README.rst

```diff
@@ -0,0 +1,100 @@
+.. _ot_ncp_sample:
+
+Thread: NCP sample
+##################
+
+The Thread NCP sample demonstrates the usage of OpenThread Network Co-Processor architecture inside the Zephyr shell.
+
+Overview
+********
+
+The sample allows to build Thread Border Router target on the MCU. 
+It is important to mention that, according to NCP architecture, MCU part
+needs to cooperate with user higher layer process to altogether establish complete 
+Thread Border Router. Recommended solution is to use docker version of OTBR (OpenThread Border Router)
+and WPAN Network Daemon tool (wpantund), working under Linux operating system, for such purpose 
+(for more information read this: https://openthread.io/guides/border-router).
+
+There is possibility to configure number of OpenThread features, but by default
+application has COMMISSIONER, JOINER, BORDER_ROUTER, BORDER_AGENT and DHCP6_SERVER
+functionalities enabled.
+
+Requirements
+************
+
+* One of the following development kits for testing the network status:
+
+  * |nRF52840DK|
+  * |nRF52833DK|
+
+* Border Router higher layer process communicating with MCU border router part.
+
+* Optionally, one or more compatible development kits programmed with OpenThread samples to test communication
+  in bigger network.
+
+User interface
+**************
+
+Main recommended interaction with the application should be done via wpantund
+Linux process using its comannds (for more information: https://github.com/openthread/wpantund).
+It is also possible to communicate with board using pyspinel tool via serial interface 
+(for more information: https://github.com/openthread/pyspinel). 
+
+Building and running
+********************
+
+.. |sample path| replace:: :file:`samples/openthread/ncp`
+
+|enable_thread_before_testing|
+
+.. include:: /includes/build_and_run.txt
+
+Testing
+=======
+
+To perform tests it is necessary to have at least one developement kit board
+and second Thread node for checking connection. Depending on needs it might be 
+another developement kit board running OpenThread stack or the simulated node on PC.
+
+Below steps were based on the given connectivity test: https://openthread.io/guides/border-router/docker/test-connectivity,
+where detailed description may be found.
+
+#. First of all it is needed to build sample and programm developement kit with it.
+#. Connect board to the PC and get its serial port name.
+#. Run OTBR using docker and pass board serial port name as ncp interface.
+#. Now bring up second Thread node, for example by flashing developement
+   kit board with OpenThread CLI FTD_JOINER sample (see here: https://codelabs.developers.google.com/codelabs/openthread-hardware/#4) 
```

Comment:
if possible please refer to some existing ncs sample instead. Note: commands below will need to be adjusted